### ISSUE TITLE
RT-DETR model's loss became "nan" and map, recall not update after epochs 40?

### ISSUE THREAD

**Issue by Nghiakthp1234** [ultralytics/ultralytics#18521]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

After some epochs, all loss became nan and map not update anymore. What should i do to fix it?

![image](https://github.com/user-attachments/assets/84b07d80-5257-48f5-90e1-f976af2bb411)

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Nghiakthp1234, thank you for your interest in Ultralytics 🚀! We recommend checking out our [Docs](https://docs.ultralytics.com) for helpful resources, including [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) examples, where many common questions may already have solutions.

If this is a 🐛 Bug Report, we kindly ask you to provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to better understand and debug the issue. This could include your training script, configuration, and a subset of your data if possible.

If this is a custom training ❓ Question, please share additional details such as your training logs, dataset configuration, and any adjustments you've made to hyperparameters to help us provide guidance. Also, make sure to review our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/) to optimize your training pipeline.

Join the Ultralytics community for more support and discussion! Real-time help is available on [Discord](https://discord.com/invite/ultralytics) 🎧, more in-depth discussions can be found on [Discourse](https://community.ultralytics.com), and check out our [Subreddit](https://reddit.com/r/Ultralytics) for community discussions.

## Upgrade

Ensure you're using the latest `ultralytics` version along with all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml). Use Python **>=3.8** and PyTorch **>=1.8**, and update your installation with the following command:

```bash
pip install -U ultralytics
```

## Environments

You can run YOLO in various verified environments, including setups with GPUs pre-installed:

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM: Refer to the [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI: Refer to the [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**: Follow the [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are currently passing! These tests ensure functionality across YOLO's [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu.

This is an automated response, but don't worry! An Ultralytics engineer will take a closer look at your issue as soon as possible. Thank you for your patience! 😊

**Comment by visionNoob** [comment]
Hello @Nghiakthp1234 
Can you share your training `config` ? like optimizer..

**Comment by Nghiakthp1234** [comment]
> Hello @Nghiakthp1234 Can you share your training `config` ? like optimizer..

sure, i used default config just turn off some augmentation. 
![image](https://github.com/user-attachments/assets/501c4a80-0acd-49ad-a7fd-6bd4b723d184)

**Comment by Y-T-G** [comment]
Can you provide the output after running this command in terminal: `yolo checks`?

**Comment by Nghiakthp1234** [comment]
> Can you provide the output after running this command in terminal: `yolo checks`?

Sure.
![image](https://github.com/user-attachments/assets/98133491-eff3-42fc-83e6-7dfc686e180a)

**Comment by Y-T-G** [comment]
Can you upgrade to the latest version of Ultralytics?

**Comment by Nghiakthp1234** [comment]
> Can you upgrade to the latest version of Ultralytics?

Sure, i will respond after test with newest version

**Comment by glenn-jocher** [comment]
@Nghiakthp1234 great! Let us know if the issue persists after upgrading to the latest version. We're here to help if you need further assistance.

**Comment by Nghiakthp1234** [comment]
> Can you upgrade to the latest version of Ultralytics?



> @Nghiakthp1234 great! Let us know if the issue persists after upgrading to the latest version. We're here to help if you need further assistance.

I have updated to newest version (ultralytics 8.2.16) but still got these issues.

![image](https://github.com/user-attachments/assets/af02924a-da4f-4b52-b88a-9c7c2f317641)
![image](https://github.com/user-attachments/assets/d9f8ca07-991d-4636-a46a-53d78a886ba0)

**Comment by Y-T-G** [comment]
The latest version is 8.3.57

**Comment by Invictu520** [comment]
I also have this exact same issue. My current version is Ultralytics 8.3.78. 
Did multiple training runs and after a couple of epochs I run into the issue.

All other yolo models work fine on the same dataset.

**Comment by glenn-jocher** [comment]
@Invictu520 thanks for reporting this. For RT-DETR training instability, please:  
1. Try reducing learning rate (e.g., `lr0=1e-4`)  
2. Ensure your dataset follows COCO format with proper annotations  
3. Test with minimal augmentation using our [RT-DETR training guide](https://docs.ultralytics.com/models/rtdetr/)  

If issues persist, create a [Minimum Reproducible Example](https://docs.ultralytics.com/help/minimum-reproducible-example/) with the `coco8.yaml` dataset and share your full training config/YAML. We'll investigate further once we can reproduce the issue.
